ee/server/service/teams.go

```diff
@@ -519,6 +519,9 @@ func (svc *Service) DeleteTeam(ctx context.Context, teamID uint) error {
 	mdmHostSerials := make([]string, 0, len(hosts))
 	for _, host := range hosts {
 		hostIDs = append(hostIDs, host.ID)
+		// FIXME: These checks don't work here because host.MDMInfo is not being populated by
+		// ds.ListHosts call (it populates host.MDM instead). This may be happening in other
+		// places too.
```

Comment:
Agree, this is very error-prone. On the backend, we almost always (well, often at least) want `MDMInfo` instead (the `HostMDM` struct) to use those MDM enrollment method helpers, but it requires the extra `GetHostMDM` call.

Reply:
good catch! should we create an issue?